micro-benchmarks/DRB200-sync1_no.c

```diff
@@ -0,0 +1,48 @@
+/*
+!!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
+!!! Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+!!! and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
+!!!
+!!! SPDX-License-Identifier: (BSD-3-Clause)
+!!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
+ */
+
+/*
+ * This is a program based on a dataset contributed by
+ * Wenhao Wu and Stephen F. Siegel @Univ. of Delaware.
+
+ * Two threads sync using one lock. No race.
+ */
+
+#include <stdio.h>
+#include <omp.h>
+#include <assert.h>
+
+omp_lock_t l;
+int x = 1;
+
+int main()
+{
+  omp_init_lock(&l);
+#pragma omp parallel num_threads(2)
+  {
+    int tid = omp_get_thread_num();
+    if (tid == 0)
+      omp_set_lock(&l);
+#pragma omp barrier
+    if (tid == 0)
+    {
+      x = 0;
+      omp_unset_lock(&l);
+    }
+    else if (tid == 1)
+    {
+      omp_set_lock(&l);
+      omp_unset_lock(&l);
```

Comment:
thread with id 0 change x's value after unsetting the lock. Would this cause data races?

Reply:
The thread with id 0 acquires the lock before the first barrier, modifies x, and then releases the lock after the first barrier. Is that right?

Reply:
I misspoke, the code is about thread with id 1. It upsets a lock, then access x.  The lock does not protect the access. Would this cause data races?